### PR TITLE
Wait for pod to be ready before connecting

### LINKED ISSUE #329: draft connect should wait for pods to come up before establishing connection
for all apps, the UX for the current `draft up` to `draft connect` has a gap: the starting of the containers. It doesn't take that long on minikube, end even shorter in the cloud with bigger machines, but I'm wondering whether it's not a better experience to check the pods for completion as part of the Building, Pushing, Releasing.... stages. Truth is, "Releasing" is really, "deploying the chart" and as a result, it returns SUCCESS when your chart is deployed but the containers are unready. This means that you are lead to _expect_ that you can immediately `draft connect`, but that fails for short while until the pods are running. 

Below, after Releasing comes back as success, on my minikube there's about ten seconds to wait before you can use `draft connect`. 

image

### PR DESCRIPTION
**Pull request by radu-matei**
This PR fixes #329 

- can someone with an available Minikube test it? At some point @michelleN did some testing and got some timeout errors - I haven't seen any issues on AKS so far

### CODE REVIEW

File: pkg/kube/podutil/podutil.go [hunk 1]
```diff
@@ -35,3 +42,30 @@ func GetPodCondition(status *v1.PodStatus, conditionType v1.PodConditionType) (i
 	}
 	return -1, nil
 }
+
+// GetPod waits for a pod with the specified label to be ready, then returns it
+// if no pod is ready, it checks every second until a pod is ready
+func GetPod(namespace, labelKey, labelValue string, clientset kubernetes.Interface) *v1.Pod {
```
Comment (bacongobbler) [inline pkg/kube/podutil/podutil.go]: Would it be better to accept a label selector instead? e.g. https://github.com/kubernetes/helm/blob/48e703997016f3edeb4f0b90e6cfdb3456ce3db0/pkg/helm/portforwarder/portforwarder.go#L58-L59

Comment (radu-matei) [reply pkg/kube/podutil/podutil.go]: Yes, it makes sense as will also add a label with the build ID later on.

**Comment by bacongobbler** [comment]
One thing I noticed was the lack of a timeout. Does that exist in this PR? If we're waiting for a pod to come ready, is it possible that this function will never exit?

**Comment by radu-matei** [comment]
Yeah, totally makes sense - how many seconds do you think is reasonable to wait for pods to be ready?
I'm thinking somewhere around 30 seconds?

**Comment by bacongobbler** [comment]
Helm has a timeout of 5 minutes which is pretty reasonable.

**Pull request merged**